###############################################################################
#
# mkfontdir 1.x Makefile
#
# Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
# Use is subject to license terms.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the "Software"),
# to deal in the Software without restriction, including without limitation
# the rights to use, copy, modify, merge, publish, distribute, sublicense,
# and/or sell copies of the Software, and to permit persons to whom the
# Software is furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice (including the next
# paragraph) shall be included in all copies or substantial portions of the
# Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
# DEALINGS IN THE SOFTWARE.
#

If mkfontdir is called from an alternate path, look there first for mkfontscale
(needed for use in packaging scripts in SVR4 packages used in Live Upgrade or
 other alternate root scenarios).

--- mkfontdir.cpp	Thu Mar  6 13:44:46 2008
+++ mkfontdir.cpp	Fri Sep 12 16:28:45 2008
@@ -1,4 +1,12 @@
 XCOMM!/bin/sh
 XCOMM $XFree86: xc/programs/mkfontdir/mkfontdir,v 1.2 2003/07/18 14:51:43 tsi Exp $
 
-exec BINDIR/mkfontscale -b -s -l "$@"
+FONTUTILDIR=`dirname $0`
+if [ ! -x ${FONTUTILDIR}/mkfontscale ] ; then
+    FONTUTILDIR=${FONTUTILDIR}/../bin
+    if [ ! -x ${FONTUTILDIR}/mkfontscale ] ; then
+        FONTUTILDIR=BINDIR
+    fi
+fi
+
+exec ${FONTUTILDIR}/mkfontscale -b -s -l "$@"
